connector-sdk/document/src/main/java/io/camunda/document/ExternalDocument.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.document;
+
+import io.camunda.client.api.response.DocumentMetadata;
+import io.camunda.document.reference.DocumentReference;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.OffsetDateTime;
+import java.util.Base64;
+import java.util.Map;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+public class ExternalDocument implements Document {
+
+  private final DocumentMetadata metadata;
+  private final HttpURLConnection connection;
+
+  public ExternalDocument(String url, @Nullable String name, HttpURLConnection connection) {
+    try {
+      if (connection == null) {
+        URL urlObj = new URL(url);
+        connection = (HttpURLConnection) urlObj.openConnection();
+        connection.setRequestMethod("GET");
+        connection.connect();
+        connection.getContentLengthLong();
+        connection.getContentType();
+      }
+      this.connection = connection;
+
+      metadata = createMetadata(name);
+    } catch (IOException e) {
```

Comment:
Shouldnt this go through the regular proxy enabled http base client?

Reply:
We should also be careful doing io heavy operations inside the constructor. If we decided to keep the whole document in memory we could do the fetching outside of this class and just pass the data as a byte[].